In [1]:
import pandas as pd
import seaborn as sns
import scipy.stats as stats
import numpy as np


In [2]:
earning=pd.read_csv('C:/earnings1.1.csv', usecols=['height', 'weight', 'male', 'earn', 'earnk', 'ethnicity', 'education', 'mother_education', 'father_education', 'walk', 'exercise', 'smokenow', 'tense', 'angry', 'age'])


In [3]:
earning

height  weight  male     earn  earnk ethnicity  education  \
0         74   210.0     1  50000.0   50.0     White       16.0   
1         66   125.0     0  60000.0   60.0     White       16.0   
2         64   126.0     0  30000.0   30.0     White       16.0   
3         65   200.0     0  25000.0   25.0     White       17.0   
4         63   110.0     0  50000.0   50.0     Other       16.0   
...      ...     ...   ...      ...    ...       ...        ...   
1811      61   120.0     0  15000.0   15.0     White       18.0   
1812      64   130.0     0   8000.0    8.0     White       12.0   
1813      72   194.0     1  60000.0   60.0     White       12.0   
1814      63   155.0     0  15000.0   15.0     Other       14.0   
1815      68   150.0     1   6000.0    6.0     White       12.0   

      mother_education  father_education  walk  exercise  smokenow  tense  \
0                 16.0              16.0     3         3       2.0    0.0   
1                 16.0              16.0     6         5       1.0    0.0   
2                 16.0              16.0     8         1       2.0    1.0   
3                 17.0               NaN     8         1       2.0    0.0   
4                 16.0              16.0     5         6       2.0    0.0   
...                ...               ...   ...       ...       ...    ...   
1811              18.0              18.0     6         1       2.0    0.0   
1812              12.0              12.0     1         1       1.0    7.0   
1813              12.0              12.0     2         1       2.0    0.0   
1814              14.0              14.0     6         1       2.0    2.0   
1815              12.0              12.0     1         6       1.0    2.0   

      angry  age  
0       0.0   45  
1       0.0   58  
2       1.0   29  
3       0.0   57  
4       0.0   91  
...     ...  ...  
1811    0.0   82  
1812    7.0   33  
1813    0.0   50  
1814    2.0   69  
1815    2.0   27  

[1816 rows x 15 columns]

In [4]:
earning.columns

Index(['height', 'weight', 'male', 'earn', 'earnk', 'ethnicity', 'education',
       'mother_education', 'father_education', 'walk', 'exercise', 'smokenow',
       'tense', 'angry', 'age'],
      dtype='object')

In [5]:
earning_cleaned = earning.dropna()

1. Considerando las variables earn, height y age, vamos a suponer que nuestro dataset
fuera una población completa. (Prepara los datos adecuadamente).
a. Realiza 5 pruebas con muestras de tamaño n = 25, en cada caso, con niveles
de significancia diferentes pero inferiores a 0,1, para confirmar o rechazar las
siguientes hipótesis considerando las alternativas. Crea para ello una fórmula
que reciba los parámetros adecuados y responda "con nivel de confianza del
...%, se acepta la hipótesis nula/alternativa".

In [6]:
import pandas as pd
from scipy import stats
def test_hipotesis(earning_cleaned , columna, valor_hipotesis, alternativa): #Definimos la función
    # Realizar la prueba t de una muestra
    t_stat, p_value = stats.ttest_1samp(earning_cleaned [columna], popmean=valor_hipotesis)

    # Determinar si se acepta o rechaza la hipótesis nula
    if alternativa == "!=":
        if p_value < 0.1:
            decision = "rechazar la hipótesis nula"
        else:
            decision = "aceptar la hipótesis nula"
    elif alternativa == ">":
        if p_value < 0.1:
            decision = "rechazar la hipótesis nula"
        else:
            decision = "aceptar la hipótesis nula"
    elif alternativa == "<":
        if p_value < 0.1:
            decision = "rechazar la hipótesis nula"
        else:
            decision = "aceptar la hipótesis nula"

    # Mostrar el resultado de la prueba
    print(f"Para la variable '{columna}':")
    print(f"Hipótesis nula (H0): x = {valor_hipotesis}")
    print(f"Hipótesis alternativa (H1): x {alternativa} {valor_hipotesis}")
    print(f"P-valor: {p_value}")
    print(f"Con un nivel de significancia de 0.1, se decide {decision}\n")

# Realizar pruebas de hipótesis para las variables especificadas
test_hipotesis(earning_cleaned, 'earn', 20000, '!=')
test_hipotesis(earning_cleaned, 'height', 55, '!=')
test_hipotesis(earning_cleaned, 'age', 44, '!=')
test_hipotesis(earning_cleaned, 'earn', 35000, '!=')
test_hipotesis(earning_cleaned, 'height', 68, '!=')
test_hipotesis(earning_cleaned, 'age', 56, '!=')
test_hipotesis(earning_cleaned, 'earn', 25000, '>')
test_hipotesis(earning_cleaned, 'age', 63, '>')
test_hipotesis(earning_cleaned, 'age', 51, '>')
test_hipotesis(earning_cleaned, 'earn', 32000, '<')
test_hipotesis(earning_cleaned, 'age', 70, '<')
test_hipotesis(earning_cleaned, 'age', 62, '<')


Para la variable 'earn':
Hipótesis nula (H0): x = 20000
Hipótesis alternativa (H1): x != 20000
P-valor: 0.00018624859604560756
Con un nivel de significancia de 0.1, se decide rechazar la hipótesis nula

Para la variable 'height':
Hipótesis nula (H0): x = 55
Hipótesis alternativa (H1): x != 55
P-valor: 0.0
Con un nivel de significancia de 0.1, se decide rechazar la hipótesis nula

Para la variable 'age':
Hipótesis nula (H0): x = 44
Hipótesis alternativa (H1): x != 44
P-valor: 1.0144654507441158e-13
Con un nivel de significancia de 0.1, se decide rechazar la hipótesis nula

Para la variable 'earn':
Hipótesis nula (H0): x = 35000
Hipótesis alternativa (H1): x != 35000
P-valor: 1.947200786708014e-79
Con un nivel de significancia de 0.1, se decide rechazar la hipótesis nula

Para la variable 'height':
Hipótesis nula (H0): x = 68
Hipótesis alternativa (H1): x != 68
P-valor: 1.4504105288807194e-38
Con un nivel de significancia de 0.1, se decide rechazar la hipótesis nula

Para la variable 'ag

b. Calcula la media poblacional para cada variable. ¿Son correctos los
resultados obtenidos por tus pruebas?

In [80]:
# Calculamos la media poblacional para cada variable
media_earn = earning_cleaned['earn'].mean()
media_height = earning_cleaned['height'].mean()
media_age = earning_cleaned['age'].mean()

# Imprimir las medias poblacionales
print(f"Media poblacional para 'earn': {media_earn}")
print(f"Media poblacional para 'height': {media_height}")
print(f"Media poblacional para 'age': {media_age}")


Media poblacional para 'earn': 22356.17638888889
Media poblacional para 'height': 66.6673611111111
Media poblacional para 'age': 40.82222222222222


Para la variable 'earn': La media poblacional calculada es aproximadamente 22356.18. La hipótesis nula especificaba un valor de 20000.
La diferencia entre la media poblacional y el valor hipotético es significativa, lo que coincide con la decisión de rechazar la hipótesis nula en las pruebas de hipótesis.

Para la variable 'height':La media poblacional calculada es aproximadamente 66.67. La hipótesis nula especificaba un valor de 55.
La diferencia entre la media poblacional y el valor hipotético también es significativa, lo que coincide nuevamente con la decisión de rechazar la hipótesis nula en las pruebas de hipótesis.

Para la variable 'age': La media poblacional calculada es aproximadamente 40.82.
La hipótesis nula especificaba un valor de 44.
En este caso, la diferencia entre la media poblacional y el valor hipotético no parece ser significativa, lo que puede indicar que las pruebas de hipótesis podrían haber rechazado incorrectamente la hipótesis nula.

2. Crea una función que estime la proporción de hombres en el dataset, considerando
una muestra de 50 individuos, con niveles de significancia de 0,05 y 0,01. Escribe con
palabras tus resultados ejecutar la función e interpreta. Compara con la proporción
real.

In [91]:

def estimar_proporcion_hombres(data , muestra_size, nivel_significancia=0.05):
    # Tomamos una muestra aleatoria de tamaño muestra_size
    muestra = earning_cleaned.sample(n=muestra_size, random_state=42)
    
    # Calcular la proporción de hombres en la muestra
    prop_hombres_muestra = muestra['male'].mean()
    
    # Calcular la proporción real de hombres en el dataset (earning_cleaned) completo
    prop_hombres_real = earning_cleaned['male'].mean()
    
    # Realizar una prueba de hipótesis para comparar la proporción de hombres en la muestra con la proporción real
    z_stat, p_value = stats.ttest_1samp(muestra['male'], prop_hombres_real)
    
    # Imprimir los resultados de la prueba de hipótesis
    print(f"Proporción de hombres en la muestra: {prop_hombres_muestra}")
    print(f"Proporción real de hombres en el dataset: {prop_hombres_real}")
    print(f"P-valor: {p_value}")
    
    # Comparar el p-valor con el nivel de significancia dado
    if p_value < nivel_significancia:
        print(f"Con un nivel de significancia de {nivel_significancia}, se rechaza la hipótesis nula.")
    else:
        print(f"Con un nivel de significancia de {nivel_significancia}, no se rechaza la hipótesis nula.")

# función con un nivel de significancia de 0.05
print("Resultados con nivel de significancia de 0.05:")
estimar_proporcion_hombres(earning_cleaned, muestra_size=50, nivel_significancia=0.05)

#  de significancia de 0.01
print("\nResultados con nivel de significancia de 0.01:")
estimar_proporcion_hombres(earning_cleaned, muestra_size=50, nivel_significancia=0.01)


Resultados con nivel de significancia de 0.05:
Proporción de hombres en la muestra: 0.38
Proporción real de hombres en el dataset: 0.37777777777777777
P-valor: 0.9745641381362697
Con un nivel de significancia de 0.05, no se rechaza la hipótesis nula.

Resultados con nivel de significancia de 0.01:
Proporción de hombres en la muestra: 0.38
Proporción real de hombres en el dataset: 0.37777777777777777
P-valor: 0.9745641381362697
Con un nivel de significancia de 0.01, no se rechaza la hipótesis nula.


La proporción en la muestra es del 38%, mientras que la proporción real de hombres en el conjunto de datos completos es aprox. del 37.78%. Por lo tanto, en ambos casos, con niveles de significancia de 0.05 y 0.01 respectivamente, no se rechaza la hipótesis nula, lo que significa que no hay evidencia sufiencte para afirmar que la proporción de hombres difiere significativamente de la proporción real en el conjunto de datos completos.

Considerando el dataset como una muestra, verifica si el género (male=1 significa
‘hombre”) influye sobre el sueldo “earn” de las personas. Explica y justifica tu
procedimiento.

In [95]:
#Lo primimemero es dividir el dataset por genero
# Dividir el dataset por género
salarios_hombres = earning_cleaned[earning_cleaned['male'] == 1]['earn']
salarios_mujeres = earning_cleaned[earning_cleaned['male'] == 0]['earn']


#Planteamos la hipótesis:
#Hipótesis nula (H0): No hay diferencia significativa en los salarios entre hombres y mujeres.
#Hipótesis alternativa (H1): Hay una diferencia significativa en los salarios entre hombres y mujeres.

# Realizar la prueba estadística (prueba t de Student para muestras independientes)
t_statistic, p_value = stats.ttest_ind(salarios_hombres, salarios_mujeres)

# Calcular la diferencia de medias entre los salarios de hombres y mujeres
diferencia_medias = salarios_hombres.mean() - salarios_mujeres.mean()

# Imprimir la magnitud de la diferencia
print("Diferencia en salarios entre hombres y mujeres:", diferencia_medias)
# Nivel de significancia típico de 0.05
nivel_significancia = 0.05

# Interpretar los resultados
if p_value < nivel_significancia:
    print("Hay una diferencia significativa en los salarios entre hombres y mujeres.")
else:
    print("No hay una diferencia significativa en los salarios entre hombres y mujeres.")



#Si hay una diferencia, esta puede ser que debido a que en general hay mas hombres, por tanto, es más probable que hayan más hombres que ganen más, por pura probabilidad al tener más opciones, esta es la opción más probable con los...
#Datos que tenemos, para un análisis más completo, deberíamos estudiar y conocer datos como: Nivel educacional,experiencia laboral,horas trabajadas, entre otros.

Diferencia en salarios entre hombres y mujeres: 15348.379858193279
Hay una diferencia significativa en los salarios entre hombres y mujeres.
